decoder.go

```diff
@@ -105,3 +111,64 @@ func (dec *Decoder) DecodeFloat32(data []byte, pcm []float32) (int, error) {
 	}
 	return n, nil
 }
+
+// Decode encoded Opus data into the supplied buffer with forward error
+// correction. The supplied buffer will be entirely filled.
+func (dec *Decoder) DecodeFEC(data []byte, pcm []int16) error {
+	if dec.p == nil {
+		return errDecUninitialized
+	}
+	if len(data) == 0 {
+		return fmt.Errorf("opus: no data supplied")
+	}
+	if len(pcm) == 0 {
+		return fmt.Errorf("opus: target buffer empty")
+	}
+
+	n := int(C.opus_decode(
+		dec.p,
+		(*C.uchar)(&data[0]),
+		C.opus_int32(len(data)),
+		(*C.opus_int16)(&pcm[0]),
+		C.int(cap(pcm)),
```

Comment:
This brought to my attention a bug in all the Decode* functions :/ I don't remember why I originally used `cap`, but I think it's a bug. It's asymmetrical with io.Reader, whose signature I tried to emulate. It was probably originally chosen for alignment with the Encode* functions and the suggestion to resize them (data = data[:n]), but it doesn't make sense for Decode.

Anyway, this is all outside the scope of your PR, so it's all good here.